packages/services/src/services/Analytics/Analytics.js

```diff
@@ -86,6 +86,81 @@ class AnalyticsAPI {
       root.removeEventListener('scroll', () => scrollAnalytics);
     }
   }
+
+  /**
+   * This instantiates an event listener to trigger an event if the Carbon
+   * Tabs component is being interacted with by the user
+   *
+   * @example
+   * import { AnalyticsAPI } from '@carbon/ibmdotcom-services';
+   *
+   * function init() {
+   *   AnalyticsAPI.initDynamicTabs();
+   * }
+   */
+  static initDynamicTabs() {
+    const fireEvent = this.registerEvent;
+
+    root.document.addEventListener('tab-selected', function(evt) {
+      try {
+        fireEvent({
+          type: 'element',
+          primaryCategory: 'WIDGET',
+          eventName: 'CLICK',
+          eventCategoryGroup: 'TABS DYNAMIC',
+          executionPath: evt.target.id,
+          targetTitle: evt.detail.item.innerText,
+        });
+      } catch (err) {
+        console.error('Error triggering tab event:', err);
+      }
+    });
+  }
+
+  /**
+   * This instantiates an event listener to trigger an event if the Carbon
+   * Modal component is being interacted with by the user
+   *
+   * @example
+   * import { AnalyticsAPI } from '@carbon/ibmdotcom-services';
+   *
+   * function init() {
+   *   AnalyticsAPI.initModals();
+   * }
+   */
+  static initModals() {
+    const fireEvent = this.registerEvent;
+
+    root.document.addEventListener('modal-hidden', function(evt) {
+      try {
+        fireEvent({
+          type: 'element',
+          primaryCategory: 'WIDGET',
+          eventName: 'HIDE',
+          eventCategoryGroup: 'SHOWHIDE',
+          executionPath: evt.target.id,
+          targetTitle: evt.detail.launchingElement.innerText,
+        });
+      } catch (err) {
+        console.error('Error triggering modal hide event:', err);
+      }
+    });
+
+    root.document.addEventListener('modal-shown', function(evt) {
```

Comment:
Does it make sense to bring these out into their methods, with more specific naming `onModalOpen`?

Reply:
I was just thinking this when reviewing the expressive modal. I think that implementation might have to call these methods directly, so have to probably break this one out.

Reply:
Updated in https://github.com/carbon-design-system/ibm-dotcom-library/pull/571/commits/421039ac7f4faef4256f586450b37ea2427a1f36